static/js/components/AllocationPage.jsx

```diff
@@ -174,11 +192,25 @@ const AllocationList = ({ allocations, deletePermission }) => {
                 root: classes.allocationDelete,
                 disabled: classes.allocationDeleteDisabled,
               }}
-              onClick={() => deleteAllocation(allocation)}
+              onClick={() => openDialog(allocation.id)}
               disabled={!deletePermission}
             >
-              &times;
+              <DeleteIcon />
             </Button>
+            <Dialog open={dialogOpen} onClose={closeDialog}>
+              <DialogTitle>Delete Allocation?</DialogTitle>
+              <DialogContent>
+                Are you sure you want to delete this allocation?
+              </DialogContent>
+              <DialogActions>
+                <Button secondary autoFocus onClick={closeDialog}>
+                  Dismiss
+                </Button>
+                <Button primary onClick={() => deleteAllocation()}>
+                  Confirm
+                </Button>
+              </DialogActions>
+            </Dialog>
```

Comment:
There's a lot of duplicated code for these dialogs
What about a new component?

Reply:
@jadalilleboe How would you imagine refactoring?

Reply:
I would imagine a ConfirmDeletionDialog component which takes the delete function,  dialog state, and name of the resource as arguments

Reply:
@jadalilleboe Ok see what you think of the new component? If it makes sense, I can add it into the others.

Reply:
@mcoughlin I like it!